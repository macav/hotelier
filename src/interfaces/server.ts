export enum Status {
  RUNNING = 'running',
  RESTARTING = 'restarting',
  STOPPED = 'stopped',
  CRASHED = 'crashed',
}
export interface Server {
  id: string;
  status: Status;
  env: { [key: string]: any };
}
