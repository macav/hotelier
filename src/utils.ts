import isElectron from 'is-electron';
import { Server, Status } from './interfaces';

export default class Utils {
  static isElectron = (): boolean => {
    return isElectron();
  };

  static openExternalLink = (url: string): void => {
    if (Utils.isElectron()) {
      window.shell.openExternal(url);
    } else {
      window.open(url);
    }
  };

  static sortServers = (servers: Server[]): Server[] => {
    return servers.sort((a, b) => {
      if (a.status === Status.RUNNING && b.status !== Status.RUNNING) {
        return -1;
      } else if (b.status === Status.RUNNING && a.status !== Status.RUNNING) {
        return 1;
      }
      if (a.id < b.id) {
        return -1;
      } else if (a.id > b.id) {
        return 1;
      }
      return 0;
    });
  };
}
