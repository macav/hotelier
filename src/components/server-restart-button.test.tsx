import { shallow } from 'enzyme';
import React from 'react';
import hotelApi from '../api';
import { Server, Status } from '../interfaces';
import utils from '../utils';
import ServerList from './server-list';
import ServerRestartButton from './server-restart-button';

describe('ServerRestartButton', () => {
  const server: Server = {
    id: 'test-server-1',
    status: Status.RUNNING,
    env: {},
  };
  let updateServerStatusFn: any;
  let loadServersFn: any;

  beforeEach(() => {
    updateServerStatusFn = jest.fn();
    loadServersFn = jest.fn();
  });

  function create(btnStyle?: string) {
    return shallow(<ServerRestartButton btnStyle={btnStyle} server={server} />);
  }

  it('matches the snapshot', () => {
    const rendered = create();
    expect(rendered).toMatchSnapshot();
  });

  it('matches snapshot with provided btnStyle', () => {
    const rendered = create('btn-primary');
    expect(rendered).toMatchSnapshot();
  });

  describe('#restartServer', () => {
    let instance: ServerRestartButton;

    beforeEach(() => {
      instance = create().instance() as ServerRestartButton;
      instance['api'].startServer = jest.fn().mockResolvedValue(true);
      instance['api'].stopServer = jest.fn().mockResolvedValue(true);
      jest.useFakeTimers();
      instance.restartServer();
    });

    afterEach(() => {
      jest.useRealTimers();
    });

    it('calls the api', () => {
      expect(instance['api'].startServer).toHaveBeenCalledWith('test-server-1');
      expect(instance['api'].stopServer).toHaveBeenCalledWith('test-server-1');
    });

    it('updates the state', () => {
      expect(instance.state.server.status).toEqual(Status.RESTARTING);
      jest.runTimersToTime(1000);
      expect(instance.state.server.status).toEqual(Status.RUNNING);
    });
  });
});
