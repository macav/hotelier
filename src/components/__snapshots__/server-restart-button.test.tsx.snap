// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerRestartButton matches snapshot with provided btnStyle 1`] = `ShallowWrapper {}`;

exports[`ServerRestartButton matches the snapshot 1`] = `ShallowWrapper {}`;
